### PR TITLE
Sentiment analysis with Huggingface transformers finetuned model 

### PR DESCRIPTION
**Pull request by sreehari-aot**
# Issue Tracking

JIRA: 
Issue Type: BUG/ FEATURE

# Changes

*New inference API which uses the fine-tuned transformer model to make predictions*
*Removed the old model and its dependencies*
*Updated the cron jobs to use the new transformers model*
*Provided support for SQL Server and Postgres databases and new DBMS of choice*
# How has the change been tested?

*Unit tests and manual verification*
# Screenshots (if applicable)

# Build Success screenshot (Till a CICD pipeline is set up)

# Notes

**New  env config added DBMS="Name of dbms"**
supported values are SQL_SERVER and POSTGRES (default)
*Model used [xaqren/sentiment_analysis](https://huggingface.co/xaqren/sentiment_analysis/tree/main)*
*For handling large data inputs truncation is enabled for the inference API which would truncate the exceeding data for inference. A better approach is using chucking of the large data, which is in progress*

### CODE REVIEW

File: forms-flow-data-analysis-api/src/api/resources/sentiment_analysis.py [hunk 1]
```diff
@@ -5,16 +5,15 @@
...
+from api.services.transformers import sentiment_analysis_pipeline_transformers
 from api.utils import auth, cors_preflight
 
 API = Namespace("sentiment", description="API endpoint for sentiment analysis")
 
-
 @cors_preflight("POST,OPTIONS")
-@API.route("", methods=["POST", "OPTIONS"])
-class SentimentAnalysisResource(Resource):
+@API.route("/predict", methods=["POST", "OPTIONS"])
```
Comment (sumesh-aot) [inline forms-flow-data-analysis-api/src/api/resources/sentiment_analysis.py:15]: any reason to change the endpoint name ?

**Review by sumesh-aot — APPROVED** [review]

File: forms-flow-data-analysis-api/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ ADD . /forms-flow-data-analysis-api/app
 RUN pip install .
 
 EXPOSE 5001
-RUN python3 -m nltk.downloader punkt vader_lexicon subjectivity
+RUN python3 -c "from transformers import pipeline; pipeline('sentiment-analysis', model='xaqren/sentiment_analysis')"
```
Comment (sumesh-aot) [inline forms-flow-data-analysis-api/Dockerfile:16]: Please make the same change in jobs Dockerfile too

Comment (sreehari-aot) [reply forms-flow-data-analysis-api/Dockerfile:16]: sure

File: forms-flow-data-analysis-api/src/api/resources/sentiment_analysis.py (hunk 1 above)
Comment (sreehari-aot) [reply forms-flow-data-analysis-api/src/api/resources/sentiment_analysis.py:15]: This was done before removing the old endpoint, will update the name to old one

**Comment by sreehari-aot** [comment]
*The build will fail until the current model weights are updated, in the meantime the following model can be used, which would work as usual*
[Working model](Seethal/Distilbert-base-uncased-fine-tuned-service-bc)

**Comment by sreehari-aot** [comment]
> _The build will fail until the current model weights are updated, in the meantime the following model can be used, which would work as usual_ [Working model](Seethal/Distilbert-base-uncased-fine-tuned-service-bc)
Update:
First model weights are updated.

File: jobs/sentiment-analysis/sentiment_analysis.py [hunk 3]
```diff
@@ -43,20 +65,21 @@ def update_sentiment():
...
 
         # Find primary key for the table.
         primary_keys = _find_primary_keys(conn, table_name)
-
+        log_info(f"found primary keys : {primary_keys}")
         # Query the rows from table.
         cols_to_query = f'{primary_keys},{input_col}'
-        rows_query = f"select {cols_to_query} from {table_name} where coalesce({output_col}, '') = ''"
-
+        rows_query = f"select {cols_to_query} from {table_name} where coalesce({output_col}, '') = '' limit 100"
```
Comment (sreehari-aot) [inline jobs/sentiment-analysis/sentiment_analysis.py:81]: Setting the limit seems causing issues in SQL Server @sumesh-aot looking into it

**Review by abhilash-aot — APPROVED** [review]

**Pull request merged**